src/L.Control.Heightgraph.js

```diff
@@ -63,10 +66,18 @@ L.Control.Heightgraph = L.Control.extend({
             this._svg.selectAll("*")
                 .remove();
         }
+
         this._data = data;
-        this._init_options();
         this._prepareData();
         this._computeStats();
+
+        this._onAddData();
+    },
+    /**
+     * Trigger a re-render of the chart based on the existing data (e.g. on container resize).
+     */
+    _onAddData() {
```

Comment:
I moved these calls out of `addData`, so that all this functionality can be called from the outside if the chart is to be re-rendered (but there is not need for full reinitialization).